.kokoro/tests/run_tests.sh

```diff
@@ -38,11 +38,44 @@ gcloud auth activate-service-account\
     --key-file=$GOOGLE_APPLICATION_CREDENTIALS \
     --project=$GOOGLE_CLOUD_PROJECT
 
-# Run the tests
+echo -e "\n******************** TESTING AFFECTED PROJECTS ********************"
+set +e
+RESULT=0
 cd github/java-docs-samples
-mvn --batch-mode --fail-at-end clean verify \
-    -Dfile.encoding="UTF-8" \
-    -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
-    -Dmaven.test.redirectTestOutputToFile=true \
-    -Dbigtable.projectID="${GOOGLE_CLOUD_PROJECT}" \
-    -Dbigtable.instanceID=instance
+# For every pom.xml (may break on whitespace)
+for file in **/pom.xml; do
+    # Navigate to project
+    file=$(dirname "$file")
+    pushd "$file" > /dev/null
+
+    # Check if Java version matches test version
+    VERSION=$(grep "source>$JAVA_VERSION" pom.xml -c)
```

Comment:
So if each POM we need the following (we can update the java11 samples to use this format over the configuration in the plugin section):

```xml
    <properties>
        <maven.compiler.target>1.11</maven.compiler.target>
        <maven.compiler.source>1.11</maven.compiler.source>
    </properties>
```

Then we can use the following: `VERSION=$(grep -oP '(?<=<maven.compiler.target>).*?(?=</maven.compiler.target>)' pom.xml)`

And then it looks like we can [check against a comma separated list](https://stackoverflow.com/questions/29300806/a-bash-script-to-check-if-a-string-is-present-in-a-comma-separated-list-of-strin) like so: 
```bash
JAVA_VERSIONS="1.8,1.11"
VERSION="1.11"
if [[ ",$JAVA_VERSIONS," = *",$VERSION,"* ]]; then
  // run tests
fi
```